static/nav-highlight.js

```diff
@@ -0,0 +1,54 @@
+// Highlight sidebar nav items based on the current scroll position.
+
+// Number of category nav items on the page
+const numCategories = document.getElementById('organized_list_items');
+const num = Number(numCategories.dataset.name);
+
+var timer = null;
+let currentElem = 1;
+
+// Find and highlight the currently scrolled section
+const findHiddenElem = () => {
+  let elemIndex = 0;
+  for (let i = 0; i < num; i += 1) {
+    const elem = document.getElementsByClassName('organized-by')[i].getBoundingClientRect();
+    if (elem.top < 60) {
+      elemIndex += 1;
+    }
+  }
+  elemIndex += 1;
+  if(currentElem !== elemIndex) {
+    document.body
+      .querySelector('.table tbody tr:nth-child(' + currentElem + ') a')
+      .removeAttribute("style");
+  }
+  document.body
+    .querySelector('.table tbody tr:nth-child(' + elemIndex + ') a')
+    .setAttribute("style", "color: #576523; font-weight: bold;");
+  currentElem = elemIndex;
+}
+
+// Run the callback `fn` at most every `ms`
```

Comment:
Move debounce logic to a function and decrease the delay, make the delay more invisible to users.